ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,235 @@
+            out_dir (Path): directory to write this dataset to
+            custom_dataset_name (str): name of the dataset's containing folder
+        """
+        dataset_path = self.dataset_path / self.dataset_name
+        print(dataset_path)
+        self.delete_dir(dataset_path) # Only deletes the contents already inside the directory
+
+        test_subset, dev_subset = self.split_data(obj_list)
+
+        # Test subset.
+        self.write_related_data(test_subset, dataset_path / 'test')
+
+        dev_path = dataset_path / 'splits'
+
+        # standard_path = dev_path / 'standard'
+        # write_out_fold(standard_path, fold, is_standard=True)
+
+        complete_path = dev_path / 'complete'
+        self.write_out_complete_set(complete_path, dev_subset)
+
+    def delete_dir(self, path):
+        """Deletes all files and folders in specified directory
+        
+        Args:
+            path (Path object): target directory
+        """
+        if os.path.isdir(path):
+            for the_file in os.listdir(path):
+                file_path = os.path.join(path, the_file)
+                try:
+                    if os.path.isfile(file_path):
+                        os.unlink(file_path)
+                    elif os.path.isdir(file_path):
+                        shutil.rmtree(file_path)
+                except Exception as e:
+                    print(e)
+                    raise Exception(
+                        "Directory deletion failed. Maybe you have a file explorer/terminal open in this dir?"
+                    )
+
+            print('\n', "Done deleting contents of", path)
+
+    def split_data(self, obj_list):
+        """Splits obj_list into test/dev sets
+        
+        Args:
+            obj_list (list): list of objects to divide into test/dev
+            test_percent (int): percentage of objects in the test set
+        
+        Returns:
+            tuple of two lists. The first list is test, second dev
+        """
+        if len(obj_list) == 0:
+            raise Exception("Empty object list passed.")
+        if len(obj_list) == 1:
+            raise Exception(
+                "Object list of length 1 passed. Can't build test and dev set with this."
+            )
+
+        shuffle(obj_list)
+        index_to_split_on = max(1, int(len(obj_list) * self.test_percent))
+
+        test = obj_list[:index_to_split_on]
+        dev = obj_list[index_to_split_on:]
+
+        return (test, dev)
+
+    def write_related_data(self, objects, path):
+        if not os.path.exists(path):
+            os.makedirs(path)
+        for obj in objects:
+            self.copy_associated_files(path, obj)
+
+    def copy_associated_files(self, destination, obj):
+            data_dir = self.temp_dir
+
+            for suffix in self.associated_files.values():
+                for prefix in self.related_data_prefixes.values():
+                    filepath = data_dir / f'{prefix}{obj.get("image_id")}{suffix}'
```

Comment:
I know this is a holdover from jigsaw, but the whole "associated_files" + "related_data_prefixes" scheme never made any sense to me. Based on this code, one is just a list of prefixes and the other a list of suffixes that are combined and used to determine the associated files. At best, these variables are just horrendously misnamed. It also makes no sense to doubly iterate over every combination of prefix and suffix.

The simplest replacement would be a list of (prefix, suffix) pairs: i.e. `[('image_', '.png'), ('image_', '.jpg'), ('meta_', '.json')]`. At least that would be somewhat readable. A more flexible idea would be to take a list of format strings, i.e. `['image_{object_id}.png', 'meta_{object_id}.json']`.

Reply:
Also see my comment above about `need_func`, this seems like it should be in sync with `load_data`.

Reply:
agree with the naming issue here - I think the original idea was to say "here are the prefixes for this type of data", and then "here are the filetypes that might be part of that type of data." we should use a clearer naming convention.